src/main/java/com/synopsys/integration/detect/workflow/blackduck/font/DetectFontInstaller.java

```diff
@@ -35,21 +42,33 @@ public void installFonts(File targetDirectory) {
             String location = artifactResolver.resolveArtifactLocation(ArtifactoryConstants.ARTIFACTORY_URL, ArtifactoryConstants.FONTS_REPO, ArtifactoryConstants.FONTS_PROPERTY, "", "");
             File fontZipFile = downloadZipFile(location, targetDirectory);
             expandZipFile(fontZipFile, targetDirectory);
+            installedToolManager.saveInstalledToolLocation(FONTS_ZIP_KEY, targetDirectory.getAbsolutePath());
+
             String zipFilePath = fontZipFile.getAbsolutePath();
             logger.debug("Deleting zip file {}", zipFilePath);
             boolean zipDeleted = FileUtils.deleteQuietly(fontZipFile);
             if (zipDeleted) {
                 logger.debug("Successfully deleted zip file {}", zipFilePath);
             }
-        } catch (IOException | IntegrationException | ArchiveException ex) {
-            logger.error(String.format("Failed to load font files into %s", targetDirectory.getAbsolutePath()));
-            logger.error("Cause: ", ex);
+        } catch (IOException | IntegrationException | ArchiveException e) {
+            logger.debug("Attempting to download fonts from previous install.");
+            installedToolLocator.locateTool(FONTS_ZIP_KEY).ifPresent(fonts ->
+            {
+                try {
+                    FileUtils.copyDirectory(fonts, targetDirectory);
+                } catch (IOException ex) {
+                    logger.error(String.format("Failed to load font files into %s", targetDirectory.getAbsolutePath()));
+                    logger.error("Online install exception cause: ", e);
+                    logger.error("Local install exception: ", ex); //TODO- should we log exceptions for both online and local installs?
```

Comment:
Probably? I think the user might want to know WHY the local version is being used. But also the local install exception would be helpful for us.

Reply:
I would consider moving the Online installer error to above "Attempting to download from previous install"